packages/wrangler/src/user.tsx

```diff
@@ -864,7 +864,12 @@ export async function loginOrRefreshIfRequired(
     // If we are not interactive, we cannot ask the user to login
     return isInteractive && (await login());
   } else if (isAccessTokenExpired()) {
-    return await refreshToken();
+    const didRefresh = await refreshToken();
+    if (didRefresh) {
+      return true;
+    } else {
+      return isInteractive && (await login());
+    }
```

Comment:
I think this could simply be...

```suggestion
    return await refreshToken() || isInteractive && (await login());
```

Reply:
That was my first take, but I landed on this version because it reads better. I'll add a comment.